pkgs/applications/blockchains/bitbox/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildNpmPackage
+, clang
+, go
+, qtwebengine
+, wrapQtAppsHook
+}:
+
+let
+  src = fetchFromGitHub {
+    owner = "digitalbitbox";
+    repo = "bitbox-wallet-app";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-kvkfz9zwleZNB+leefx+cciJCPkFSSRDxBNRKNqeODc=";
+  };
+  version = "4.40.0";
+  web = buildNpmPackage {
+    pname = "bitbox-web";
+    inherit version;
+    inherit src;
+    sourceRoot = "source/frontends/web";
+    npmDepsHash = "sha256-bnMmeSX8UZpndHK0NMLQhX1GSunO0JSyZdUTfG+rSpY=";
+    installPhase = "cp -r build $out";
+  };
+in
+stdenv.mkDerivation {
+  pname = "bitbox";
+  inherit version;
+
+  inherit src;
+
+  patches = [
+    ./desktop.patch
+    ./genassets.patch
+  ];
+
+  dontConfigure = true;
+
+  buildPhase = ''
+    ln -s ${web} frontends/web/build
+    export GOCACHE=$TMPDIR/go-cache
+    make -C frontends/qt/server linux
+    make -C frontends/qt base
+  '';
+
+  installPhase = ''
+    mkdir $out
+    cp -r frontends/qt/resources/linux/usr/share $out
+    mkdir $out/{bin,lib}
+    cp frontends/qt/build/BitBox $out/bin/bitbox
+    cp frontends/qt/build/assets.rcc $out/bin
+    cp frontends/qt/server/libserver.so $out/lib
+    install -Dt $out/lib/udev/rules.d ${./rules.d}/*
+  '';
+
+  buildInputs = [ qtwebengine ];
+  nativeBuildInputs = [
+    clang
+    go
+    wrapQtAppsHook
+  ];
+
+  meta = with lib; {
+    description = "Companion app for the BitBox02 hardware wallet";
+    homepage = "https://bitbox.swiss/app/";
+    license = licenses.asl20;
+    maintainers = [ maintainers.tensor5 ];
```

Comment:
I remember reading that it defaults to the package name.

Reply:
yes but:
> trace: warning: getExe: Package "qdmr-0.11.3" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, use getExe' to specify the name to the program, such as lib.getExe' foo "bar".

![image](https://github.com/NixOS/nixpkgs/assets/80165193/be865944-1819-4439-9ab3-d925f295f46c)

Reply:
fba19fadf06ca6ca7c49d0986e37b6cb074d31d0